Update docker-compose.yml for a static config file, and update traefik examples (#6142)
matrix-org/synapse: contrib/docker/README.md

# Synapse Docker

### Configuration

A sample ``docker-compose.yml`` is provided, including example labels for
reverse proxying and other artifacts. The docker-compose file is an example,
please comment/uncomment sections that are not suitable for your usecase.

Specify a ``SYNAPSE_CONFIG_PATH``, preferably to a persistent path,
to use manual configuration. 

To generate a fresh `homeserver.yaml`, you can use the `generate` command. 
(See the [documentation](../../docker/README.md#generating-a-configuration-file) 
for more information.) You will need to specify appropriate values for at least the 
`SYNAPSE_SERVER_NAME` and `SYNAPSE_REPORT_STATS` environment variables. For example:

```
docker-compose run --rm -e SYNAPSE_SERVER_NAME=my.matrix.host -e SYNAPSE_REPORT_STATS=yes synapse generate
```

> Above command contains variable SYNAPSE_REPORT_STATS which is opt-in only. If you say 'yes' you will help project with some anonymous usage details, when you opt for 'no' it will disable such functionality. This variable is mandatory (see ../../docker/README.md).  

Reviewer: I'm sorry, but I still don't think this is very clear. 

1. "Above command" is incorrect; it should be "The above command"
2. "opt-in only" is unclear.
3. The link is broken
4. please break markdown lines at 80 characters or so to make it easier to read and review
5. There should be a blank line between paragraphs.

But utimately I think it duplicates information which is already available in the linked documentation or in the paragraph above and is redundant. Please just remove it.

Author: Fixed per your suggestion, however even if it duplicates in some part, it easier to configure docker container i think. Personally, i was surprised i need this variable (and i didn't read it about it in this particular doc page), i had it in my synapse configuration, but during migrating to docker i simply didn't remember about it.

> https://github.com/matrix-org/synapse/issues/5518. Contributions to bring it up

Reviewer: do you think it is now sufficiently up-to-date to remove this paragraph?

Author: yes, i checked this by running docker-compose with almost any variables. As written in the issue, one should go with configuration to homeserver.yaml (however i was surprised opt-in for stats was left here).

Reviewer: ok, so please can you remove this paragraph?

Author: done, forgot about this.

> to use manual configuration. To generate a fresh ``homeserver.yaml``, simply run:

Reviewer: ```suggestion
to use manual configuration. 

To generate a fresh `homeserver.yaml`, you can use the `generate` command. 
(See the [documentation](../../docker/README.md#generating-a-configuration-file) 
for more information.) You will need to specify appropriate values for at least the 
`SYNAPSE_SERVER_NAME` and `SYNAPSE_REPORT_STATS` environment variables. For example:
```

```
docker-compose run --rm -e SYNAPSE_SERVER_NAME=my.matrix.host -e SYNAPSE_REPORT_STATS=yes synapse generate
```

> Above command contains variable SYNAPSE_REPORT_STATS which is opt-in only. You are free to say 'no' here. 

Reviewer: Currently this reads:

> You are free to say 'no' here. This will also generate necessary signing keys. 

... which is misleading. I'm suggesting you reword this by adding some more text above the example.

```suggestion
```

Author: I have made some changes here.

Reviewer 2: Still not sure where the stats are sent.  Is this simply telling the client to send some status to their home synapse server?

> This will also generate necessary signing keys.

Reviewer: ```suggestion
(This will also generate necessary signing keys.)
```

to use manual configuration. To generate a fresh ``homeserver.yaml``, simply run:

```

> docker-compose run --rm -e SYNAPSE_SERVER_NAME=my.matrix.host e SYNAPSE_REPORT_STATS=yes synapse generate

Reviewer: we should not tell people to set `SYNAPSE_REPORT_STATS=yes` without understanding what it does: can you add some words to explain that this is an *example*.

> docker-compose run --rm -e SYNAPSE_SERVER_NAME=my.matrix.host e SYNAPSE_REPORT_STATS=yes synapse generate

Reviewer: Also:

```suggestion
docker-compose run --rm -e SYNAPSE_SERVER_NAME=my.matrix.host -e SYNAPSE_REPORT_STATS=yes synapse generate
```

Author: its fixed
